sbt/src/test/scala/testpkg/ServerSpec.scala

```diff
@@ -47,6 +47,60 @@ class ServerSpec extends AsyncFreeSpec with Matchers {
         (s contains """"id":11""") && (s contains """"error":""")
       })
     }
+
+    "return basic completions on request" in withTestServer("events") { p =>
+      p.writeLine(
+        """{ "jsonrpc": "2.0", "id": 15, "method": "textDocument/completion", "params": { } }"""
```

Comment:
See my comment in https://github.com/sbt/sbt/issues/4374#issuecomment-424921945.

Reply:
I understand the ultimate goal of your proposal and make sense.
but I'm not sure we should go in that direction from day 0 ... having contextual completion is great for IDEs, not sure how much added value it has in this context considering the "lag" it will produce...